### ISSUE TITLE
Couldn't connect to 127.0.0.1 on port 2003, is carbon-agent.py running?

### ISSUE THREAD

**Issue by philippe-tr** [pkkummermo/grafana-vagrant-puppet-box#10]
Performed the following steps after having run successfully and issued `vagrant halt`:
```
$ vagrant up
$ vagrant provision
$ vagrant ssh
$ cd /opt/graphite/examples
$ python example-client.py
```
Received error message: `Couldn't connect to 127.0.0.1 on port 2003, is carbon-agent.py running?`

**Comment by pkkummermo** [comment]
Seems as if there are some unstability to wheter carbon starts or not. Try this as a temporary solution:
// In project folder
vagrant ssh
sudo service carbon restart

**Comment by philippe-tr** [comment]
I issued:
    vagrant ssh
    sudo service carbon restart
The response was
     * Restarting carbon database carbon                                                                   [ OK ]
I then issued
    ps aux | grep carbon
The response was
    vagrant   2686  0.0  0.0  10464   920 pts/0    R+   14:42   0:00 grep --color=auto carbon

**Comment by pkkummermo** [comment]
Ah so it's been up before? 

Do a dump of
```
tail --l 500 /var/log/carbon/console.log
```
and we'll get to the bottom of it.

There also might be that there's a stale pid-file just lying around as well :)

**Comment by philippe-tr** [comment]
Thanks. Here is what I get from the tail command:

```
18/02/2015 13:54:15 :: /opt/graphite/conf/storage-aggregation.conf not found, ignoring.
# many more entries just like this removed for brevity
18/02/2015 14:29:06 :: /opt/graphite/conf/storage-aggregation.conf not found, ignoring.
18/02/2015 14:29:17 :: Received SIGTERM, shutting down.
18/02/2015 14:29:17 :: Carbon shutting down.  Update rate not changed
18/02/2015 14:29:17 :: (TCP Port 7002 Closed)
18/02/2015 14:29:17 :: Stopping factory <twisted.internet.protocol.ServerFactory instance at 0x7f18a0c64878>
18/02/2015 14:29:17 :: (TCP Port 2004 Closed)
18/02/2015 14:29:17 :: Stopping factory <twisted.internet.protocol.ServerFactory instance at 0x7f18a0c64a28>
18/02/2015 14:29:17 :: (TCP Port 2003 Closed)
18/02/2015 14:29:17 :: Stopping factory <twisted.internet.protocol.ServerFactory instance at 0x7f18a0c68e60>
18/02/2015 14:29:18 :: Main loop terminated.
18/02/2015 14:29:18 :: Server Shut Down.
```

**Comment by pkkummermo** [comment]
Doesn't seem like Carbon has tried to start. The service might return [OK] even though the restart haven't been triggered.

Try to hit this one:
```
sudo service carbon restart && tail -f /var/log/carbon/console.log
```
Also, check for other suspicious logs (error/stdout) in /var/log/carbon. Can't remember what the different logs are named of the top of my head.

Also, the missing /opt/graphite/conf/storage-aggregation.conf has no effect on carbon. :)

**Comment by philippe-tr** [comment]
```
sudo service carbon restart && tail -f /var/log/carbon/console.log
 * Restarting carbon database carbon                                                                   [ OK ]
18/02/2015 14:29:17 :: Received SIGTERM, shutting down.
18/02/2015 14:29:17 :: Carbon shutting down.  Update rate not changed
18/02/2015 14:29:17 :: (TCP Port 7002 Closed)
18/02/2015 14:29:17 :: Stopping factory <twisted.internet.protocol.ServerFactory instance at 0x7f18a0c64878>
18/02/2015 14:29:17 :: (TCP Port 2004 Closed)
18/02/2015 14:29:17 :: Stopping factory <twisted.internet.protocol.ServerFactory instance at 0x7f18a0c64a28>
18/02/2015 14:29:17 :: (TCP Port 2003 Closed)
18/02/2015 14:29:17 :: Stopping factory <twisted.internet.protocol.ServerFactory instance at 0x7f18a0c68e60>
18/02/2015 14:29:18 :: Main loop terminated.
18/02/2015 14:29:18 :: Server Shut Down.
```

```
/var/log/carbon$ ls
console.log  console.log.2015_2_17  creates.log  query.log
```

```
/opt/graphite/storage$ ls
graphite.db  index  lists  log  rrd  whisper
```

**Comment by pkkummermo** [comment]
Any pid-file?

**Comment by philippe-tr** [comment]
There was no /opt/graphite/storage/carbon-cache.pid. 
I also looked in var/run.
```
/var/run$ ls | grep pid
acpid.pid
acpid.socket
atd.pid
crond.pid
dhclient.eth0.pid
rpc.statd.pid
rsyslogd.pid
sm-notify.pid
sshd.pid
upstart-file-bridge.pid
upstart-socket-bridge.pid
upstart-udev-bridge.pid
```

At this point I think I will start over with the image, then snapshot it. "vagrant suspend" is safe, but "vagrant halt" causes this situation where carbon does not restart.

**Comment by pkkummermo** [comment]
General tip:
All data is stored as files @ 
```
/opt/graphite/storage/whisper/
```
Copy it to /vagrant and you'll be able to paste them into the new box :)

Still seems like a issue worth checking out, even though I've halted my Vagrant box severalt times with no problems. Created a new issue #12 to keep a "clean" issue story.

**Comment by philippe-tr** [comment]
I found an error I introduced that caused the `sudo service carbon restart` not to work. I had added a new section to puppet\modules\carbon\files\storage-schemas.conf and had a typo. I corrected the typo and re-ran `vagrant provision`.

I was then able to run `sudo service carbon restart` successfully. So I can now do this:
```
$ vagrant halt
$ vagrant up
$ vagrant ssh
$ sudo service carbon restart
```

Thanks for your outstanding support.
